exonum/src/api/node/private/mod.rs

```diff
@@ -0,0 +1,213 @@
+
+#[doc(hidden)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct ServiceInfo {
+    name: String,
+    id: u16,
+}
+
+/// `DTO` is used to transfer information about node.
+#[doc(hidden)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct NodeInfo {
+    pub core_version: Option<String>,
+    pub protocol_version: u8,
+    pub services: Vec<ServiceInfo>,
+}
+
+impl NodeInfo {
+    /// Creates new `NodeInfo`, from services list.
+    pub fn new<'a, I>(services: I) -> NodeInfo
+    where
+        I: IntoIterator<Item = &'a Box<Service>>,
+    {
+        let core_version = option_env!("CARGO_PKG_VERSION").map(|ver| ver.to_owned());
+        NodeInfo {
+            core_version,
+            protocol_version: PROTOCOL_MAJOR_VERSION,
+            services: services
+                .into_iter()
+                .map(|s| ServiceInfo {
+                    name: s.service_name().to_owned(),
+                    id: s.service_id(),
+                })
+                .collect(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Default)]
+struct ReconnectInfo {
+    delay: u64,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "type")]
+enum IncomingConnectionState {
+    Active,
+    Reconnect(ReconnectInfo),
+}
+
+impl Default for IncomingConnectionState {
+    fn default() -> IncomingConnectionState {
+        IncomingConnectionState::Active
+    }
+}
+
+#[derive(Serialize, Deserialize, Default)]
+struct IncomingConnection {
+    public_key: Option<PublicKey>,
+    state: IncomingConnectionState,
+}
+
+#[derive(Serialize, Deserialize)]
+struct PeersInfo {
+    incoming_connections: Vec<SocketAddr>,
+    outgoing_connections: HashMap<SocketAddr, IncomingConnection>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+struct PeerAddQuery {
+    ip: SocketAddr,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+struct ConsensusEnabledQuery {
+    enabled: bool,
+}
+
+/// Private system API.
+#[derive(Clone, Debug)]
+pub struct SystemApi {
+    info: NodeInfo,
+    shared_api_state: SharedNodeState,
+}
+
+impl SystemApi {
+    /// Creates a new `private::SystemApi` instance.
+    pub fn new(info: NodeInfo, shared_api_state: SharedNodeState) -> SystemApi {
+        SystemApi {
+            info,
+            shared_api_state,
+        }
+    }
+
+    /// Adds private system API endpoints to the corresponding scope.
+    pub fn wire(self, api_scope: &mut ServiceApiScope) -> &mut ServiceApiScope {
+        self.handle_peers_info("v1/peers", api_scope)
+            .handle_peer_add("v1/peers", api_scope)
+            .handle_network_info("v1/network", api_scope)
+            .handle_is_consensus_enabled("v1/consensus_enabled", api_scope)
+            .handle_set_consensus_enabled("v1/consensus_enabled", api_scope)
+            .handle_shutdown("v1/shutdown", api_scope);
+        api_scope
+    }
+
+    fn handle_peers_info(self, name: &'static str, api_scope: &mut ServiceApiScope) -> Self {
```

Comment:
This signature uses for clarify code in the `wire` method.

```rust
        self.handle_peers_info("v1/peers", api_scope)
            .handle_peer_add("v1/peers", api_scope)
            .handle_network_info("v1/network", api_scope)
            .handle_is_consensus_enabled("v1/consensus_enabled", api_scope)
            .handle_set_consensus_enabled("v1/consensus_enabled", api_scope)
            .handle_shutdown("v1/shutdown", api_scope);
        api_scope
```

instead of

```
        self.handle_peers_info(api_scope)
            .handle_peer_add(api_scope)
            .handle_network_info(api_scope)
....
        api_scope
```